src/common/guac_surface.c

```diff
@@ -665,19 +909,20 @@ guac_common_surface* guac_common_surface_alloc(guac_client* client,
         guac_socket* socket, const guac_layer* layer, int w, int h) {
 
     /* Init surface */
-    guac_common_surface* surface = malloc(sizeof(guac_common_surface));
+    guac_common_surface* surface = calloc(1, sizeof(guac_common_surface));
```

Comment:
For the automatic zeroing of memory. From the `calloc()` section of the `malloc()` man page:

> ### NAME
> 
> malloc, free, calloc, realloc - allocate and free dynamic memory
> 
> ### SYNOPSIS
>     #include <stdlib.h>
> 
>     void *malloc(size_t size);
>     void free(void *ptr);
>     void *calloc(size_t nmemb, size_t size);
>     void *realloc(void *ptr, size_t size);

> ### DESCRIPTION
> 
> ...
> 
> The `calloc()`  function allocates memory for an array of `nmemb` elements of size bytes each and returns a pointer to the allocated memory. **The memory  is  set  to zero.**  If `nmemb` or `size` is 0, then `calloc()` returns either `NULL`, or a unique pointer value that can later  be  successfully passed to `free()`.
> 
> ...
> 

Emphasis added. No such guarantee in normal `malloc()`. This is often faster than `malloc()` + `memset()`, as the kernel keeps a zeroed pages of memory available, and `calloc()` can exploit this. More on that: http://stackoverflow.com/questions/2688466/why-mallocmemset-is-slower-than-calloc